neutron_policy_server/wsgi.py

```diff
@@ -16,40 +19,85 @@
 app = Flask(__name__)
 
 
-@app.route("/enforce", methods=["POST"])
-def enforce():
-    data = request.json
-    rule = data.get("rule")
-    target = data.get("target")
-    creds = data.get("creds")
-
+def _fetch_context(request):
+    content_type = request.headers.get('Content-Type', default="application/x-www-form-urlencoded")
+    if content_type == "application/x-www-form-urlencoded":
+        data = request.form.to_dict()
+    elif content_type == "application/json":
+        data = request.json
+    target = json.loads(data.get("target")) if (
+            type(data.get("target")) is str) else data.get("target")
+    creds = json.loads(data.get("credentials")) if (
+            type(data.get("credentials")) is str) else data.get("credentials")
     ctx = context.Context(user_id=creds["user_id"], project_id=creds["project_id"])
+    return target, ctx
```

Comment:
Let's replace this with a [`before_request`](https://flask.palletsprojects.com/en/3.0.x/api/#flask.Flask.before_request) which is a bit more clean, and then let's store it into `flask.g` which is explained here:

https://flask.palletsprojects.com/en/2.3.x/api/#flask.g

That way, we don't have to call it manually each time, instead it just runs before every request and loads up the target and context automatically :)